internal/db/block/manager.go

```diff
@@ -20,82 +20,68 @@ func NewManager(database db.Database) *Manager {
 
 // GetBlockByHash search the block with the given block hash. If the block does
 // not exist then returns nil. If any error happens, then panic.
-func (manager *Manager) GetBlockByHash(blockHash types.BlockHash) *types.Block {
+func (manager *Manager) GetBlockByHash(blockHash types.BlockHash) (*types.Block, error) {
 	// Build the hash key
 	hashKey := buildHashKey(blockHash)
 	// Search on the database
 	rawResult, err := manager.database.Get(hashKey)
 	if err != nil {
-		panic(any(err))
-	}
-	// Check not found
-	if rawResult == nil {
-		// notest
-		return nil
+		return nil, err
 	}
 	// Unmarshal the data
 	block, err := unmarshalBlock(rawResult)
 	if err != nil {
-		panic(any(err))
+		return nil, err
 	}
-	return block
+	return block, nil
 }
 
 // GetBlockByNumber search the block with the given block number. If the block
 // does not exist then returns nil. If any error happens, then panic.
-func (manager *Manager) GetBlockByNumber(blockNumber uint64) *types.Block {
+func (manager *Manager) GetBlockByNumber(blockNumber uint64) (*types.Block, error) {
 	// Build the number key
 	numberKey := buildNumberKey(blockNumber)
 	// Search for the hash key
 	hashKey, err := manager.database.Get(numberKey)
 	if err != nil {
-		panic(any(err))
-	}
-	// Check not found
-	if hashKey == nil {
-		// notest
-		return nil
+		return nil, err
 	}
 	// Search for the block
 	rawResult, err := manager.database.Get(hashKey)
 	if err != nil {
-		panic(any(err))
-	}
-	// Check not found
-	if rawResult == nil {
-		// notest
-		return nil
+		return nil, err
 	}
 	// Unmarshal the data
 	block, err := unmarshalBlock(rawResult)
 	if err != nil {
-		panic(any(err))
+		return nil, err
 	}
-	return block
+	return block, nil
 }
 
 // PutBlock saves the given block with the given hash as key. If any error happens
 // then panic.
-func (manager *Manager) PutBlock(blockHash types.BlockHash, block *types.Block) {
+func (manager *Manager) PutBlock(blockHash types.BlockHash, block *types.Block) error {
 	// Build the keys
 	hashKey := buildHashKey(blockHash)
 	numberKey := buildNumberKey(block.BlockNumber)
 	// Encode the block as []byte
 	rawValue, err := marshalBlock(block)
 	if err != nil {
-		panic(any(err))
+		return err
 	}
 	// TODO: Use transaction?
 	// Save (hashKey, block)
 	err = manager.database.Put(hashKey, rawValue)
```

Comment:
The only return value is an error so again I suggest the `if err = ...` idiom here 😅.

Reply:
In this case I need the `rawValue` below and I don't want to use an else statement there